### ISSUE TITLE
executor times out waiting for docker pull of large images

### ISSUE THREAD

**Issue by jdef** [mesosphere/kubernetes-mesos#492]
there's a hard-coded 5m timeout (`launchGracePeriod = 5 * time.Minute`) in executor.go and this is breaking the experience for some users that have huge docker images that can take a while to pull. this value should be configurable.

**Comment by s-urbaniak** [comment]
xref https://github.com/kubernetes/kubernetes/pull/14432

**Comment by sttts** [comment]
There should be a `--executor-launch-grace-period` flag in the scheduler as well, passed down to the kubelet.

**Comment by hrishimantri** [comment]
@jdef when is the next release planned ? The last release I see released was 29 days back : https://github.com/mesosphere/kubernetes/releases.

**Comment by jdef** [comment]
We actually discussed this today and decided that we're still lacking some
important bits around the automation/CI story that should be in place prior
to supporting regular, short-term releases. We also decided that in the
meantime that critical fixes could be backported to release branches
ad-hoc, and that we'd "cut a release" if there was an urgent need.

The set of changes to support executor pull/launch timeouts is particularly
small and should be relatively pain-free to backport to the release branch.
